src/Adapter/PlatformServices.WinUI/Services/WinUIFileOperations.cs

```diff
@@ -108,7 +109,14 @@ public void DisposeNavigationSession(object navigationSession)
         /// </returns>
         public string GetFullFilePath(string assemblyFileName)
         {
-            return (SafeInvoke<string>(() => Path.Combine(Windows.ApplicationModel.Package.Current.InstalledLocation.Path, assemblyFileName)) as string) ?? assemblyFileName;
+            string packagedPath()
+            {
+                return SafeInvoke<string>(() => Path.Combine(Windows.ApplicationModel.Package.Current.InstalledLocation.Path, assemblyFileName)) as string ?? assemblyFileName;
```

Comment:
`InstalledLocation` == `StorageFolder`. This is a painful way to get the path

You want to use `GetPackagePathByFullName2(...PackagePathType_Effective...)`

Win11 adds `InstalledPath` returning a `string` instead of `StorageFolder` - more efficient and reliable.

Better to call [GetPackagePath()](https://docs.microsoft.com/windows/win32/api/appmodel/nf-appmodel-getpackagepath), first appearing in Win8

But that's still flawed as it only gets the 'installed' location. It doesn't support Mutable or External Location. GetPackagePathByFullName2 solves that.

Pseudo-code you want:

```
try
{
    path = GetPackagePathByFullName2(...PackagePathtype_Effective...)
}
catch (FunctionNotFound e)
{
    try
    {
        path = GetPackagePath()
    }
    catch (FunctionNotFound e)
    {
        return assemblyFileName;
    }
}
```